etc/kayobe/kolla.yml

```diff
@@ -320,9 +333,13 @@ kolla_build_blocks:
                -e '/\[{{ repo.tag }}\]/,/^\[/ s|^\(name.*\)|\1\nbaseurl={{ repo.url }}|' /etc/yum.repos.d/{{ repo.file }}{% if not loop.last %} &&{% endif %} \
     {% endfor %}
     {% endif %}
+  # With the UCA keyring installed we can now add all repos.
   base_ubuntu_package_sources_list: |
+    {% if stackhpc_kolla_clean_up_repo_mirrors | bool %}
+    COPY sources.list.ubuntu /etc/apt/sources.list.backup
+    {% endif %}
     RUN \
-        mv /etc/apt/sources.list /etc/apt/sources.list.backup && \
+        rm /etc/apt/sources.list && \
```

Comment:
:disappointed:

Reply:
There's definitely some magic going on here...

In base_header I'm replacing sources.list with a file containing the base Ubuntu repos. This is because we don't yet have the necessary keyring to use the UCA repo.

In base_ubuntu_package_sources_list we have the UCA keyring, so we can remove sources.list and build it up again including UCA.

Regarding the backup, Kolla provides the `docker/base/sources.list.ubuntu` file which is used by default on Ubuntu. I'm using it as the backup sources.list file on L339 that will be copied back into place at the end of the build in the `footer` section.